src/Microsoft.AspNetCore.SystemWebAdapters/HttpApplicationState.cs

```diff
@@ -0,0 +1,223 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace System.Web;
+
+[SuppressMessage("Design", "CA1010:Generic interface should also be implemented", Justification = Constants.ApiFromAspNet)]
+[SuppressMessage("Design", "CA1001:Types that own disposable fields should be disposable", Justification = "This is internally disposed")]
+public sealed class HttpApplicationState : NameObjectCollectionBase
```

Comment:
I'm assuming this is a straight port.

Reply:
Yup - cleaned up a little to use more modern locking primitive, but overall it's the same